eos_data_distribution/avahi_monitor.py

```diff
@@ -79,23 +96,102 @@ def __init__(self):
         check_call(
             ["nfdc", "set-strategy", str(SUBSCRIPTIONS_SOMA), 'ndn:/localhost/nfd/strategy/multicast'])
 
-        self._registry = dict()
+        self.update_routed_status()
+        self.process_route_changes(False)
+
+    def update_routed_status(self):
+        self._routed = set(
+            self._ndn_gateways.values()).intersection(self.gateways)
```

Comment:
Will this ever set `self._routed` to `True`? If I understand correctly, `set(self._ndn_gateways.values())` is a set of IP address strings, and `self.gateways` is a dict of address family to a list of gateway information tuples*. How can they ever have an intersection?

* For example, from [here](https://pypi.python.org/pypi/netifaces#how-do-i-use-it): `{2: [('10.0.1.1', 'en0', True), ('10.2.1.1', 'en1', False)], 30: [('fe80::1', 'en0', True)], 'default': { 2: ('10.0.1.1', 'en0'), 30: ('fe80::1', 'en0') }}`